src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -171,7 +171,19 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
           return searchFilters.some(getTexts => {
             const sourceTexts: string[] = [getTexts(item)].flat().map(normalizeText);
 
-            return sourceTexts.some(source => searchTexts.some(search => source.includes(search)));
+            return searchTexts.some(search => {
+              if(search.startsWith("regex:")) {
+                const matcher = new RegExp(search.substring(6).trim());
+
+                return sourceTexts.some(source => matcher.test(source));
+              } else if(search.indexOf("*") !== -1) {
```

Comment:
I personally feel that this should be removed.

Because this doesn't look obvious, it feels like a mix of "normal string" and "RegExp"

If it's for ease of input, I feel we should provide a button to selectively turn on whether to search in RegExp mode.

Just like idea:

<img width="1018" alt="图片" src="https://user-images.githubusercontent.com/8198408/121111306-35719e00-c841-11eb-8f19-7f98f89707ed.png">

Reply:
I was thinking about Googles advanced search queries like site:..., filletype:..., etc. You really don't often use them - that's why they are kept hidden. On the other hand.. Lens is for advanced users. There is not that big of a need to keep everything as simple as the google search.

Another interesting idea would be to autocomplete those advanced queries and allow the user to chain them(of course we would have to support more complex ones one the future). Like "regex:prometheus-[0-9] memory:>500M". A bit like Gitlab does it 
![Bildschirmfoto von 2021-06-08 09-09-33](https://user-images.githubusercontent.com/5812715/121139682-50401480-c839-11eb-912d-77b61118e34c.png)

Reply:
@Nokel81 @jakolehm @nevalla what do you think? Should I remove the "regex:" support or leave it in for now?

Reply:
I mean remove `*` , Instead of `regex:`

Reply:
Oh really? I think a simple glob search is much more intuitive than regex in this case. We definitely need more opinions on this :D

Reply:
The main thing is that here, there will be a problem, as I said before: 

> The main thing is that here, there will be a problem, as I said before: xxx

I think it might be better to use `glob:` instead of `*`

Reply:
Of course, I feel it would be better if a few buttons were provided

Reply:
As a long-term Linux user, I'm kinda used to globs working everywhere. Nevertheless - I get your point. I'd love to hear a couple more opinions on this. In the end, we should find the best compromise between functionality, usability, and simplicity.

Reply:
I think that just having glob search would be an improvement.

Furthermore, I would prefer having a UI and having structured search for more complex searching than trying to invent a new language.

So, if this PR could focus on adding glob support (which covers some of the regex case) I think it would be merged more quickly.

Reply:
I also think that simple glob search is more intuitive for most users. As @Nokel81 said, we could have just glob search in this PR and then follow-up issue/PR about regex.